contracts/mint/AnnualCappedInflation.sol

```diff
@@ -0,0 +1,48 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.19;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IDCNTMintAuthorization, IERC20 } from "./IDCNTMintAuthorization.sol";
+
+/// @notice an implementation of IDCNTMintAuthorization that limits mint requests to a capped annual inflation
+contract AnnualCappedInflation is Ownable, IDCNTMintAuthorization {
+    IERC20 public token;
+    uint128 public nextMint; // Timestamp
+    uint32 public constant MINIMUM_MINT_INTERVAL = 365 days;
+    uint8 public constant MINT_CAP_BPS = 200; // 2%
+
+    error MintExceedsMaximum();
+    error MintTooSoon();
+
+    constructor(IERC20 _token, uint128 _nextMint, address _owner) {
+        token = _token;
+        nextMint = _nextMint;
+        _transferOwnership(_owner);
+    }
+
+    /// @notice mint can be called at most once every 365 days, and with an amount no more than 2% of the current supply
+    /// @param amount amount of tokens to mint
+    function canMint(address, uint256 amount) public view returns (bool) {
+        if (amount > (token.totalSupply() * MINT_CAP_BPS) / 10000) {
+            revert MintExceedsMaximum();
+        }
+
+        if (block.timestamp < nextMint) {
+            revert MintTooSoon();
+        }
+
```

Comment:
I _think_ maybe if the owner of this contract is the token contract, and `canMint` is changed to `onlyOwner`, you could remove the `authorizeMint` function and put the `nextMint = uint128(block.timestamp + MINIMUM_MINT_INTERVAL);` line here, removing the two step mint process.

Reply:
I wouldn't say it's a two _step_ mint process, it's just that execution does jump between those two functions. It's still just one call to `authorizeMint` from the token contract though.

But you're right, this could totally be done in one function.

The reason I split it in two, is because I wanted to separate
- checking if a mint can happen right now (which is the `canMint` `view` function, no state changes in here)
- actually updating the state which blocks near-term future attempts to mint (which happens in the `authorizeMint` `external` function, state changes happen in here)

This made testing a little cleaner, and I like the idea of an external party being able to "check" without actually "doing it". So, I'm going to leave as is, assuming I understood your comments correctly :)